code/src/components/GameRoom.js

```diff
@@ -0,0 +1,167 @@
+//Here we fetch the data from the 'action' URL
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { gameDirection } from "../reducers/reusable";
+import { GameOver } from "./GameOver";
+import { game } from "../reducers/game";
+import { Button } from "../lib/Buttons";
+
+import light from "../img/light.jpg";
+import cavern from "../img/cavern.jpg";
+import bridge from "../img/bridge.jpg";
+import stone from "../img/stone.jpg";
+import robot from "../img/robot.jpg";
+import chest from "../img/chest.jpg";
+import content from "../img/content.jpg";
+
+import styled from "styled-components/macro";
+
+export const GameRoom = () => {
+  const dispatch = useDispatch();
+
+  const gameData = useSelector((state) => state.game.game);
+  const userName = useSelector((state) => state.game.name);
+  const arrayGame = useSelector((state) => state.game.game.actions);
+  const historyGame = useSelector((state) => state.game.history);
+
+  const onGameDirection = (direction) => {
+    dispatch(gameDirection(userName, direction));
+  };
+
+  const onHistoryBack = () => {
+    dispatch(game.actions.historyGoBack());
+  };
+
+  const gameImages = () => {
+    if (gameData.coordinates === "0,0") {
+      return cavern;
+    } else if (gameData.coordinates === "0,1") {
+      return robot;
+    } else if (gameData.coordinates === "0,2") {
+      return chest;
+    } else if (gameData.coordinates === "0,3") {
+      return light;
+    } else if (gameData.coordinates === "1,0") {
+      return bridge;
+    } else if (gameData.coordinates === "1,1") {
+      return stone;
+    } else if (gameData.coordinates === "1,3") {
+      return content;
+    }
+  };
+
+  const endGame = () => {
+    if (gameData.coordinates === "1,3") {
+      return <GameOver />;
+    } else {
+      return <BackButton
+      type="button"
+      onClick={onHistoryBack}
+      disabled={historyGame.length === 1}
+      background={historyGame.length === 1 ? "grey" : "rgb(32, 8, 191)"}
+      title={"back"}/>;
+    }
+  };
+
+  const chooseDirection = () => {
+    if (gameData.coordinates !== "1,3") {
+      return (
+        <Description>CHOOSE DIRECTION:</Description>
+      )
+    }
+  }
+
+  return (
+    <GameImage style={{ backgroundImage: `url(${gameImages()})` }}>
+      <Overlay />
+      <Description>{gameData.description}</Description>
+      {chooseDirection()}
+      <Wrapper>
+        {arrayGame &&
+          arrayGame.map((item) => {
+            return (
+              <Container key={item.description}>
+                <Button
+                  onClick={() => onGameDirection(item.direction)}
+                  title={`GO ${item.direction}`}
+                />
+                <p>{item.description}</p>
+              </Container>
+            );
+          })}
+      </Wrapper>
+      {endGame()}
```

Comment:
Not sure if you meant to:
1. Have the `Wrapper` inside the conditional 
2. Include the endgame() render inside the `Wrapper`

The final room has a small white pixel caused by the border from `Wrapper`
![image](https://user-images.githubusercontent.com/807352/103339126-d6814f00-4a80-11eb-85ea-9bfb73aa733a.png)